bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/WDChannels.java

```diff
@@ -51,43 +56,101 @@ public synchronized WorkerDataChannel getWorkerDataChannel(String name) {
         return channel;
     }
 
-    public Object tryTakeData(Strand strand, List<String> channels) throws Throwable {
+    public Object receiveDataMultipleChannels(Strand strand, ReceiveField[] receiveFields, Type targetType)
+            throws Throwable {
+        if (strand.mapResult == null) {
+            strand.mapResult = ValueCreator.createMapValue(targetType);
```

Comment:
Rather than creating maps for each worker receive can use same map and clean and reuse?

Reply:
It is possible that the target type can be different in each send-receive pair. So, we need to create new maps.